mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -364,11 +391,12 @@ private boolean isTransient(
     return isTransient;
   }
 
-  private Set<String> getTransientPropertyNames(TypeElement classElement) {
-    TransientProperties transientProperties = classElement.getAnnotation(TransientProperties.class);
+  private Set<String> getTransientPropertyNames(Set<TypeElement> typeHierarchy) {
+    Optional<ResolvedAnnotation<TransientProperties>> annotation =
+        getClassAnnotation(TransientProperties.class, typeHierarchy);
 
-    return transientProperties != null
-        ? Sets.newHashSet(transientProperties.value())
+    return annotation.isPresent()
+        ? Sets.newHashSet(annotation.get().getAnnotation().value())
```

Comment:
Isn't that Set has always one element? Why do we need to return set here?

Reply:
the annotation value is an array, so there can be multiple elements.